packages/tldraw/src/lib/utils/text/richText.ts

```diff
@@ -89,6 +90,23 @@ export function renderHtmlFromRichTextForMeasurement(editor: Editor, richText: T
 	return `<div class="tl-rich-text">${html}</div>`
 }
 
+// A weak cache used to store plaintext that's been extracted from rich text.
+const plainTextFromRichTextCache = new WeakCache<TLRichText, string>()
+
+export function isEmptyRichText(richText: TLRichText) {
+	if (richText.content.length === 1) {
+		if (!(richText.content[0] as any).content) return true
+	}
+	return false
+}
+
+export function isSingleCharacterRichText(richText: TLRichText) {
+	if (richText.content.length === 1) {
+		if ((richText.content[0] as any)?.content?.length === 1) return true
+	}
+	return false
+}
```

Comment:
these are fine for us, but we should double-check they work with non-texty extensions people might add like images, mentions, or horizontal rules